strapi/strapi
# Fix createTelemetry crash on non-shell hosts

### What does it do?

This PR adds a try/catch block to all places where `deviceId` is generated with `machineIdSync()`. As `machineIdSync` requires a shell to be available in order to work, it crashes the startup script on non-shell hosts, such as distroless containers. Instead, if no shell is available, `deviceId` will be generated by using `uuid/v4`. Although such id is not persistent, it will still help us to track analytics from a container session while the container is up and running; and allow our tracker to function without any changes. Currently, if there is no `deviceId` on the request to the tracker, such request will not go through.

### Why is it needed?

Fixes the issue where startup script would fail on non-shell hosts.

### How to test it?

Create a distroless container and run the application. It should start up as normal.

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/8893
https://github.com/strapi/strapi/issues/9920
